tests/integ/test_queries.py

```diff
@@ -13,7 +13,7 @@ def test_query_with_projection(engine):
         projection={User.email, User.username})
 
     result = query.one()
-    assert not hasattr(result, "profile")
+    assert getattr(result, "profile") is None
```

Comment:
This is a slightly weaker assertion; since unit tests should codify the contract, this should be explicit in checking `is None` or `not hasattr`.